ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -103,6 +104,24 @@ public MessageInfo(StoreKey key, long size, short accountId, short containerId,
    */
   public MessageInfo(StoreKey key, long size, boolean deleted, boolean ttlUpdated, long expirationTimeInMs, Long crc,
       short accountId, short containerId, long operationTimeMs) {
+    this(key, size, deleted, ttlUpdated, expirationTimeInMs, crc, accountId, containerId, operationTimeMs, (short) -1);
+  }
+
+  /**
+   * Construct an instance of MessageInfo.
+   * @param key the {@link StoreKey} associated with this message.
+   * @param size the size of this message in bytes.
+   * @param deleted {@code true} if the message is deleted, {@code false} otherwise
+   * @param ttlUpdated {@code true} if the message's ttl has been updated, {@code false} otherwise
+   * @param expirationTimeInMs the time at which the message will expire. A value of -1 means no expiration.
+   * @param crc the crc associated with this message. If unavailable, pass in null.
+   * @param accountId accountId of the blob
+   * @param containerId containerId of the blob
+   * @param operationTimeMs operation time in ms
+   * @param updateVersion update version of update
+   */
+  public MessageInfo(StoreKey key, long size, boolean deleted, boolean ttlUpdated, long expirationTimeInMs, Long crc,
```

Comment:
Add a test to test this constructor directly? Remember to cover `getUpdateVersion()`.

Reply:
added